V4_MIGRATION_GUIDE.md

```diff
@@ -94,23 +94,30 @@ See [the Getting Started section](https://github.com/auth0/nextjs-auth0/tree/mai
 By default, **the middleware does not protect any routes**. To protect a page, you can use the `getSession()` handler in the middleware, like so:
 
 ```ts
-export async function middleware(request: NextRequest) {
-  const authRes = await auth0.middleware(request)
+export async function middleware(request) {
+    const authRes = await auth0.middleware(request); //Returns a NextResponse object
 
-  // authentication routes — let the middleware handle it
-  if (request.nextUrl.pathname.startsWith("/auth")) {
-    return authRes
-  }
+    // Let Auth0 handle authentication-specific routes (e.g. /auth/login, /auth/callback, etc.)
+    if (request.nextUrl.pathname.startsWith("/auth")) {
```

Comment:
Also, I think this example isn't entirly accurate. 

`auth/profile` and `auth/access-token` aren't public routes.